File: java-api-core/src/com/kinvey/java/store/CallableAsyncPullRequestHelper.java
package com.kinvey.java.store;

import com.kinvey.java.Query;
import com.kinvey.java.network.NetworkManager;

import java.util.concurrent.Callable;

public class CallableAsyncPullRequestHelper implements Callable {

    private final NetworkManager.Get pullRequest;
    private final Query query;

    CallableAsyncPullRequestHelper(NetworkManager.Get pull, Query query) {
        this.pullRequest = pull;
        this.query = query;
    }

    @Override
    public PullTaskResponse call() throws Exception {
        return new PullTaskResponse(pullRequest.execute(), query);
    }
}

File: java-api-core/src/com/kinvey/java/store/CallableAsyncPullRequestHelper.kt
package com.kinvey.java.store

import com.google.api.client.json.GenericJson
import com.kinvey.java.Query
import com.kinvey.java.network.NetworkManager

import java.util.concurrent.Callable

class CallableAsyncPullRequestHelper<T: GenericJson> internal constructor(private val pullRequest: NetworkManager.Get<T>?,
                                                                          private val query: Query?)
    : Callable<PullTaskResponse<T>> {

    @Throws(Exception::class)
    override fun call(): PullTaskResponse<T> {
        return PullTaskResponse(pullRequest?.execute(), query)
    }
}
